src/python_testing/TC_CADMIN_1_11.py

```diff
@@ -0,0 +1,197 @@
+            TestStep(10, "TH_CR2 reads the CurrentFabricIndex attribute from the Operational Credentials cluster and saves as th2_idx"),
+            TestStep(11, "TH_CR1 sends the RemoveFabric command to the DUT with the FabricIndex set to th2_idx"),
+        ]
+
+    def generate_unique_random_value(self, value):
+        while True:
+            random_value = random.randint(10000000, 99999999)
+            if random_value != value:
+                return random_value
+
+    async def CommissionAttempt(
+            self, setupPinCode: int):
+
+        logging.info("-----------------Commissioning with TH_CR2-------------------------")
+        await self.th2.CommissionOnNetwork(
+            nodeId=self.dut_node_id, setupPinCode=setupPinCode,
+            filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=self.discriminator)
+
+    def pics_TC_CADMIN_1_11(self) -> list[str]:
+        return ["CADMIN.S"]
+
+    @async_test_body
+    async def test_TC_CADMIN_1_11(self):
+        self.step(1)
+
+        # Establishing TH1 and TH2 controllers
+        self.th1 = self.default_controller
+        self.discriminator = random.randint(0, 4095)
+        th2_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        th2_fabric_admin = th2_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=self.th1.fabricId + 1)
+        self.th2 = th2_fabric_admin.NewController(nodeId=2, useTestCommissioner=True)
+
+        self.step(2)
+        GC_cluster = Clusters.GeneralCommissioning
+        attribute = GC_cluster.Attributes.BasicCommissioningInfo
+        duration = await self.read_single_attribute_check_success(endpoint=0, cluster=GC_cluster, attribute=attribute)
+        self.timeout = duration.maxCumulativeFailsafeSeconds
+
+        self.step(3)
+        self.count = 0
+        params = await self.OpenCommissioningWindow(self.th1, 0x00)
+        setupPinCode = params.setupPinCode
+
+        self.step(4)
+        await self.CommissionAttempt(setupPinCode)
+
+        self.step(5)
+        await self.OpenCommissioningWindow(self.th1, 0x00)
+
+        self.step(6)
+        await self.OpenCommissioningWindow(self.th1, 0x02)
+
+        self.step(7)
+        await self.OpenCommissioningWindow(self.th2, 0x02)
+
+        self.step(8)
+        revokeCmd = Clusters.AdministratorCommissioning.Commands.RevokeCommissioning()
+        await self.th1.SendCommand(nodeid=self.dut_node_id, endpoint=0, payload=revokeCmd, timedRequestTimeoutMs=6000)
+        # The failsafe cleanup is scheduled after the command completes, so give it a bit of time to do that
+        sleep(1)
+
+        self.step(9)
+        AC_cluster = Clusters.AdministratorCommissioning
+        features = await self.read_single_attribute(dev_ctrl=self.default_controller, node_id=self.dut_node_id,
+                                                    endpoint=0, attribute=AC_cluster.Attributes.FeatureMap)
```

Comment:
Can you omit the endpoint here please? It will default to matter_test_config.endpoint, with a fallback to 0. This isn't wrong right now (administrator commissioning is always on 0), but one of the new features is going to have this cluster land on other endpoints, so we need to be able to run this on non-zero endpoints starting at SVE.

Reply:
So I did this and the code complained that the endpoint was a necessary var needed for the read_single_attribute function, so I changed to using the read_single_attribute_check_success function, however since there is only one feature for the AdministratorCommissioning cluster, if its not supported then it returns a 0 or None value, which the read_single_attribute_check_success function does not appear to like and it complains the following:
`mobly.signals.TestFailure: Details=Error reading <class 'chip.clusters.Objects.AdministratorCommissioning'>:<class 'chip.clusters.Objects.AdministratorCommissioning.Attributes.FeatureMap'> = ValueDecodeFailure(TLVValue=None, Reason=InteractionModelError(<Status.UnsupportedCluster: 195>)), Extras=None`
So due to the above I then changed to using an if else statement so if there were features on the device then it would do the read_single_attribute_check_success function, else we set the features var to the direct output from AC_cluster.Attributes.FeatureMap.value.

Reply:
The error there is saying that the cluster is unsupported, which means the endpoint is not 0. Can you post the TH logs there so we can check what's going on? FeatureMap should always be present and not error, so if it's missing, that's probably a sign there's something either wrong on the test or on the app.

Reply:
This appears to have been resolved with an update to both the TH and the virtual environment established in WSL on my local workstation. My apologies for the above issue mentioned.